### PR TITLE
Support `capabilities: none` in YML - Fix #14

### LINKED ISSUE #14: YML enhancement - Support capabilities: none
**Description**

For a less surprising developer experience `capabilities: none` could be supported to mirror the `capabilities: all` setting.

**Steps to reproduce the issue:**

Add `capabilities: none` to a LinuxKit YML file.

**Describe the results you received:**

Invalid for target.

**Describe the results you expected:**

Acts in the same way as if nothing was supplied. If none is one of the options then this should be mutually exclusive with any other option.

cc @justincormack 

I saw the schema supports "strings" - so the error is probably not coming from the validator. 

https://github.com/moby/tool/blob/master/cmd/moby/schema.go

I think this would change - to mirror the 'all' behaviour:

https://github.com/moby/tool/blob/master/cmd/moby/config.go#L312

Happy to take a stab at this change myself.

### PR DESCRIPTION
**Pull request by alexellis**
Fixes https://github.com/moby/tool/issues/14

How I tested:

I edited the `sshd` example changing `all` to `none` - this built successfully where as prior to that it would throw an error.

### CODE REVIEW

File: cmd/moby/config.go [hunk 1]
```diff
@@ -309,46 +309,51 @@ func ConfigInspectToOCI(image *MobyImage, inspect types.ImageInspect) ([]byte, e
...
-			"CAP_SYS_PACCT",
-			"CAP_SYS_PTRACE",
-			"CAP_SYS_RAWIO",
-			"CAP_SYS_RESOURCE",
-			"CAP_SYS_TIME",
-			"CAP_SYS_TTY_CONFIG",
-			"CAP_WAKE_ALARM",
+	if len(caps) == 1 {
+		cap := strings.ToLower(caps[0])
+		if cap == "none" {
```
Comment (justincormack) [inline cmd/moby/config.go]: switch statement is nicer here.

**Comment by rneugeba** [comment]
Could you update the git commit message to be more decriptive? "Fix #14" means very little when looking at the git commit log

**Comment by justincormack** [comment]
Yes, as this repo will not exist soon, there is no point referencing the issue number at all.

**Comment by alexellis** [comment]
Commit message updated.

**Comment by rneugeba** [comment]
The github interface is a bit crap, but it also looks like this is not signed-off-by

**Comment by justincormack** [comment]
it has the odd "alexellis committed with alexellis" in the API which normally means the signed off by is wrongly done.

**Comment by alexellis** [comment]
Switching over to the `switch` statement.

**Comment by alexellis** [comment]
For some reason this Mac is giving me issues with signing. I've not had to do anything special to use `-S` before - any ideas? The commit with alex ellis is probably squashing the commits into one.

```
error: gpg failed to sign the data
fatal: failed to write commit object
```

**Comment by justincormack** [comment]
we only need `-s` not `-S`...

File: cmd/moby/config.go [hunk 2]
```diff
@@ -309,46 +309,51 @@ func ConfigInspectToOCI(image *MobyImage, inspect types.ImageInspect) ([]byte, e
...
-			"CAP_SYS_RAWIO",
-			"CAP_SYS_RESOURCE",
-			"CAP_SYS_TIME",
-			"CAP_SYS_TTY_CONFIG",
-			"CAP_WAKE_ALARM",
+	if len(caps) == 1 {
+		switch cap := strings.ToLower(caps[0]); cap {
+		case "none":
+			caps = []string{}
+        case "all":
```
Comment (justincormack) [inline cmd/moby/config.go]: needs `go fmt` to fix indent

Comment (alexellis) [inline cmd/moby/config.go]: I'll fix that this evening. Otherwise is it ok now?

Comment (justincormack) [inline cmd/moby/config.go]: yes looks ok now

**Comment by alexellis** [comment]
`gofmt` applied + re-pushed.

**Pull request merged**